torchvision/prototype/transforms/_utils.py

```diff
@@ -39,14 +39,27 @@ def get_image_dimensions(image: Union[PIL.Image.Image, torch.Tensor, features.Im
     return channels, height, width
 
 
-def has_any(sample: Any, *types: Type) -> bool:
+def _parse_types_or_checks(
+    types_or_checks: Tuple[Union[Type, Callable[[Any], bool]], ...]
+) -> List[Callable[[Any], bool]]:
+    return [
+        cast(Callable[[Any], bool], lambda obj, typ=type_or_check: isinstance(obj, typ))
+        if isinstance(type_or_check, type)
+        else type_or_check
+        for type_or_check in types_or_checks
+    ]
+
+
+def has_any(sample: Any, *types_or_checks: Union[Type, Callable[[Any], bool]]) -> bool:
     flat_sample, _ = tree_flatten(sample)
-    return any(issubclass(type(obj), types) for obj in flat_sample)
+    checks = _parse_types_or_checks(types_or_checks)
+    return any(any([check(obj) for obj in flat_sample]) for check in checks)
```

Comment:
Nit: Similar to above, I think you can make this easier to read and potentially faster with loops (because you can `break` early.